x-pack/metricbeat/module/meraki/device_health/device_network_appliance_vpn_sitetosite.go

```diff
@@ -0,0 +1,83 @@
+package device_health
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	meraki_api "github.com/meraki/dashboard-api-go/v3/sdk"
+)
+
+func getNetworkApplianceVPNSiteToSite(client *meraki_api.Client, networks *meraki_api.ResponseOrganizationsGetOrganizationNetworks) (map[NetworkID]*meraki_api.ResponseApplianceGetNetworkApplianceVpnSiteToSiteVpn, error) {
+
+	networkVPNSiteToSites := make(map[NetworkID]*meraki_api.ResponseApplianceGetNetworkApplianceVpnSiteToSiteVpn)
+
+	for _, network := range *networks {
+
+		networkVPNSiteToSite, res, err := client.Appliance.GetNetworkApplianceVpnSiteToSiteVpn(network.ID)
+		if err != nil {
+			//Error: "This endpoint only supports MX networks"
+			// We just ignore this error but do not append to the list
+			if !(strings.Contains(string(res.Body()), "MX network")) {
+				//Any other problem we are going to return an error
+				return nil, fmt.Errorf("Appliance.GetNetworkApplianceVpnSiteToSiteVpn failed; [%d] %s. %w", res.StatusCode(), res.Body(), err)
+			}
+		} else {
+			networkVPNSiteToSites[NetworkID(network.ID)] = networkVPNSiteToSite
+
+		}
+
+	}
+
+	return networkVPNSiteToSites, nil
+}
+
+func reportNetwrokApplianceVPNSiteToSite(reporter mb.ReporterV2, organizationID string, devices map[Serial]*Device, networkVPNSiteToSites map[NetworkID]*meraki_api.ResponseApplianceGetNetworkApplianceVpnSiteToSiteVpn) {
+	metrics := []mapstr.M{}
+
+	for network_id, networkVPNSiteToSite := range networkVPNSiteToSites {
+		for _, device := range devices {
+			if device.NetworkID == string(network_id) {
+				metric := mapstr.M{
+					"device.address":      device.Address,
+					"device.firmware":     device.Firmware,
+					"device.imei":         device.Imei,
+					"device.lan_ip":       device.LanIP,
+					"device.location":     device.Location,
+					"device.mac":          device.Mac,
+					"device.model":        device.Model,
+					"device.name":         device.Name,
+					"device.network_id":   device.NetworkID,
+					"device.notes":        device.Notes,
+					"device.product_type": device.ProductType,
+					"device.serial":       device.Serial,
+					"device.tags":         device.Tags,
+				}
+				metric["network.appliance.vpn.site_to_site.network_id"] = network_id
+				metric["network.appliance.vpn.site_to_site.mode"] = networkVPNSiteToSite.Mode
+
+				if networkVPNSiteToSite.Mode != "none" {
+					if networkVPNSiteToSite.Hubs != nil {
+						for k, hub := range *networkVPNSiteToSite.Hubs {
+							metric[fmt.Sprintf("network.appliance.vpn.site_to_site.hub.%d.hub_id", k)] = hub.HubID
```

Comment:
there's something fishy going on with these - we have %d modifiers for non numerical types, and we shouldn't be using high cardinality fields like IDs in field names regardless.

Reply:
Fixed the for loops to remove the %d, I thought I had removed that from earlier comments, but had missed this one, should be fixed on next push (later tonight)